### ISSUE TITLE
A hvcat related issue

### ISSUE THREAD

**Issue by GiggleLiu** [FluxML/Zygote.jl#211]
A bug related to matrix construction in the mater branch, a MWE provided by @under-Peter
```julia console
julia> using Zygote

julia> function foo1(k)
           M = [1k 3k; 2k 4k]
           return M[1]
       end
foo1 (generic function with 1 method)

julia> function foo2(k)
           M = [1k 3k; 2k 4k]
           return M[2]
       end
foo2 (generic function with 1 method)

julia> function foo3(k)
           M = [1k 3k; 2k 4k]
           return M[3]
       end
foo3 (generic function with 1 method)

julia> function foo4(k)
           M = [1k 3k; 2k 4k]
           return M[4]
       end
foo4 (generic function with 1 method)

julia> num_grad(K::Real, f; δ::Real = 1e-5) = (f(K+δ/2) - f(K-δ/2))/δ
num_grad (generic function with 1 method)

julia> num_grad(0.4, foo1)
0.9999999999954489

julia> num_grad(0.4, foo2)
1.9999999999908977

julia> num_grad(0.4, foo3)
2.9999999999974487

julia> num_grad(0.4, foo4)
3.9999999999817955

julia> Zygote.gradient(foo1, 0.4)
(1.0,)

julia> Zygote.gradient(foo2, 0.4)
(3.0,)

julia> Zygote.gradient(foo3, 0.4)
(2.0,)

julia> Zygote.gradient(foo4, 0.4)
(4.0,)
```

**Comment by GiggleLiu** [comment]
Related PR #210

**Comment by MikeInnes** [comment]
Closed with #210 I think.
